### ISSUE TITLE
Thanks! How can I buy you a coffee/beer?

### ISSUE THREAD

**Issue by chonz0** [residentsummer/watoi#9]
I've used this tool yesterday and was very helpful. Can I buy you a coffee/beer? (PayPal or something alike)

**Comment by residentsummer** [comment]
Hi @chonz0! Glad that my tool helped you and thanks for your kind offer. I don't have an active PayPal account, unfortunately. But you can try sending me a [gift voucher](https://www.unixstickers.com/index.php?route=account/voucher) and I'll buy a sticker for my laptop :)
Email is my username at gmail.

**Comment by sTywin** [comment]
Thank you @residentsummer ; I can't believe this capability is not built into the app. I converted from whatever the current Android version is to 2.18.92 on iOS12 mostly successfully. The most difficult part was getting an unencrypted `msgstore.db` off my Android phone.

I saw a few of these:
`CoreData: warning: Unable to load class named 'WAChatSession' for entity 'WAChatSession'.  Class not found, using default NSManagedObject instead.
CoreData: warning: Unable to load class named 'WAGroupInfo' for entity 'WAGroupInfo'.  Class not found, using default NSManagedObject instead.
CoreData: warning: Unable to load class named 'WAGroupMember' for entity 'WAGroupMember'.  Class not found, using default NSManagedObject instead.`

And a few of these:
`2018-10-03 14:40:00.041 watoi[10020:750834] null text detected: { <body omitted>`

But all in all it appears that most of my messages made it through alive.

**Comment by hayschan** [comment]
Do you think you'll be available to write a guide?  The readme walkthrough seems a bit tough for me...such as I'm not sure how to decrypt the msgstore file (even after Googling).

**Comment by ketansoni** [comment]
@sTywin  - How did you get your msgstore.db decrypted? Did you root your android phone to get the key to decrypt msgstore.db? any reference would be appreciated.

**Comment by corupta** [comment]
Hi!
Successfully migrated my messages just now, thanks for providing this tool.
* Obtained `msgstore.db` via [YuvrajRaghuvanshiS/WhatsApp-Key-Database-Extractor](https://github.com/YuvrajRaghuvanshiS/WhatsApp-Key-Database-Extractor) (Failed in first try, but it eventually worked after playing around with its settings)
* Note: I could suggest others to verify that `msgstore.db` is fine before installing and verifying WhatsApp on iPhone. To do so, install [SQLite Browser](https://sqlitebrowser.org/) and open `msgstore.db` with it. (Do not modify msgstore or keep an unmodified copy)
* Then, Installed Whatsapp on iPhone, activated Whatsapp on iPhone.
* Took a backup via Finder (backups are taken from Finder and not iTunes nowadays)
* From a random website found through googling, downloaded Whatsapp.ipa matching the version in iPhone
* Then, went through the steps in [Readme](https://github.com/residentsummer/watoi#step-by-step-guide).

Also, the new backup I took was named `00008101-001A346914C2001E` (taken with Big Sur) a little different unlike previous one `c65314de2651deeb83cab271b8c70b598df458f4` (taken with Mojave) but it did not cause any problems.

So, to sum up, it worked flawlessly on April, 2021 with 
* macOS Big Sur 11.2.3, 
* iPhone 12 mini, iOS 14.4.2
* WhatsApp Messenger version 2.21.71.1

PS: I am also willing to buy you a coffee and the sticker gift voucher link did not work for me.

**Comment by TickingFeather** [comment]
Hello, just using this thread to report another success. I migrated from an old Galaxy Core phone to an iPhone 8, and watoi worked flawlessly on first attempt. The trickiest part was getting all the prerequisites done:

- Xcode wouldn't install on macOS Catalina, had to update my VM to Big Sur 
- For the `msgstore.db`,  I gave up on WhatsApp-Key-Database-Extractor because I couldn't get it to work, instead transferred the Whatsapp backup through GDrive to a rooted Xperia phone I had lying around, then copied the msgstore.db file straight from `/data/data/com.whatsapp/databases/`
- I got the `.ipa` file from the "Whatsapp++" alternative app on iosninja.io, as advised by #35. It worked even though it was v2.19 instead of 2.21
- Small gotcha: to transfer the messages to my rooted phone using GDrive, I had to re-activate Whatsapp on the old phone, then on the rooted one, which deactivated it on the iPhone. I already had the iOS backup, but after the final step of restoring the modified backup, I hit a Whatsapp protection that prevents repeated activations (seems iOS exclusive?), and had to wait 6 hours before trying again.
So plan ahead and avoid activating Whatsapp several times on your iPhone in the process, you may run into this.

Specs:

- Source: Samsung Galaxy Core 4G - Android 4.2.2 - Whatsapp 2.21.14.24
- (Intermediary rooted phone) Sony Xperia Z1 Compact - Android 5.1.1 - Whatsapp 2.21.14.24
- Target: iPhone 8 - iOS 14.7.1 - Whatsapp 2.21.141.1
- Mac: KVM virtual machine (disguised as 2017 iMac Pro) - macOS Big Sur 11.5.1

So thanks a lot for your work, this is really an awesome project and a great alternative to the many proprietary solutions out there. I'm interested in contributing to support media files, but I can't promise I'll have time this year.
(also, the Unixstickers gift link is dead indeed)

**Comment by schinj** [comment]
**Successfully migrated from Android to iOS!! 🥳 **
Thanks for your contribution. 👏 ❤️ 

Just wanted to mention this tool also for easy prerequisites: https://github.com/YuvrajRaghuvanshiS/WhatsApp-Key-Database-Extractor 👍

**Comment by aravindputrevu** [comment]
This worked with slight modifications. Thanks to everyone who are helping fix this walled garden mess :)

**Comment by marvinchin** [comment]
@aravindputrevu Do you recall which version of iOS/macOS you were using? I tried it with macOS Monterey 12.2.1 and iOS 15.2.1, but it didn't work. I was able to complete all the steps and restore my phone from the modified backup, but my messages were still missing after signing back into WA.

**Comment by residentsummer** [comment]
@marvinchin have you checked archive? There were reports that imported chats sometimes end up there.

**Comment by aravindputrevu** [comment]
@marvinchin I moved to Big Sur, iOS - 15.1. Please try searching in WhatsApp.
